fortnite_api/shop.py

```diff
@@ -1,266 +1,272 @@
-        self.giftable: bool = data['giftable']
-        self.refundable: bool = data['refundable']
-        self.sort_priority: int = data['sortPriority']
-        self.layout_id: str = data['layoutId']
-        self.layout: Optional[ShopLayout] = ShopLayout(data['layout']) if data['layout'] else None
-        self.dev_name: str = data['devName']
-        self.offer_id: str = data['offerId']
-        self.display_asset_path: Optional[str] = data['displayAssetPath']
-        try:
-            self.tile_size: ShopTileSize = ShopTileSize(data['tileSize'])
-        except ValueError:
-            # This is a fix, because Epic believes that they should spell "normal" as "nomal"
-            self.tile_size: ShopTileSize = ShopTileSize.NORMAL
-        self.new_display_asset_path: Optional[str] = data['newDisplayAssetPath']
-        self.new_display_asset: Optional[ShopNewDisplayAsset] = ShopNewDisplayAsset(data['newDisplayAsset']) if data[
-            'newDisplayAsset'] else None
-        self.items: List[Union[BrCosmetic, CarCosmetic, InstrumentCosmetic, LegoKitCosmetic, JamTrack]] = []
-        if data['brItems']:
-            self.items.extend([BrCosmetic(item) for item in data['brItems']])
-        if data['cars']:
-            self.items.extend([CarCosmetic(item) for item in data['cars']])
-        if data['instruments']:
-            self.items.extend([InstrumentCosmetic(item) for item in data['instruments']])
-        if data['legoKits']:
-            self.items.extend([LegoKitCosmetic(item) for item in data['legoKits']])
-        if data['tracks']:
-            self.items.extend([JamTrack(item) for item in data['tracks']])
-        self.raw_data: dict = data
-
-
-class ShopBundle:
-
-    def __init__(self, data: dict):
+from .abc import Hashable
+from .asset import Asset
+from .cosmetics import CosmeticBr, CosmeticCar, CosmeticInstrument, CosmeticLegoKit, CosmeticTrack
+from .http import HTTPClientT
+from .material import MaterialInstance
+from .utils import get_with_fallback, parse_time
+
+
+class ShopEntryBundle(Generic[HTTPClientT]):
+    """Represents a shop entry bundle.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        The name of the bundle.
+    info: :class:`str`
+        The information of the bundle.
+    image: :class:`Asset`
+        The image of the bundle.
+    """
+
+    __slots__: Tuple[str, ...] = ('name', 'info', 'image')
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
         self.name: str = data['name']
         self.info: str = data['info']
-        self.image_url: str = data['image']
-        self.raw_data: dict = data
+        self.image: Asset[HTTPClientT] = Asset(url=data['image'], http=http)
 
 
-class ShopBanner:
+class ShopEntryBanner:
+    """A class to represent a shop entry banner.
 
-    def __init__(self, data: dict):
+    Attributes
+    ----------
+    value: :class:`str`
+        The content of the banner.
```

Comment:
I think "The content of the banner." or "The text of the banner." might be better here